src/identity/caclient.rs

```diff
@@ -93,3 +125,97 @@ impl CertificateProvider for CaClient {
         Ok(certs)
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use std::time::Duration;
+
+    use super::{CaClient, CertificateProvider, IstioCertificateService};
+    use crate::{
+        identity::{Error, Identity},
+        tls,
+        xds::istio::ca::{IstioCertificateRequest, IstioCertificateResponse},
+    };
+    use async_trait::async_trait;
+
+    #[derive(Clone, Default)]
+    struct MockCertService {
+        response: Option<IstioCertificateResponse>,
```

Comment:
purely a nitpick but this could probably just be a single `Result`

Reply:
That's what I wanted to do, but `Result` isn't clone and MockCertService must be clone :/

We could look at https://docs.rs/mockall/latest/mockall/ to give us an implementation instead of writing our own struct.